File: check_cuda.py
import cv2

def check_opencv_cuda_support():
    # Check if OpenCV is compiled with CUDA support
    if cv2.cuda.getCudaEnabledDeviceCount() == 0:
        print("OpenCV was not built with CUDA support.")
    else:
        print("OpenCV was built with CUDA support.")

if __name__ == "__main__":
    check_opencv_cuda_support()

File: check_cuda.cpp
#include <opencv2/opencv.hpp>
#include <opencv2/cudaarithm.hpp>

int main() {
    // Check if OpenCV is compiled with CUDA support
    if (cv::cuda::getCudaEnabledDeviceCount() == 0) {
        std::cout << "OpenCV was not built with CUDA support." << std::endl;
    } else {
        std::cout << "OpenCV was built with CUDA support." << std::endl;
    }

    return 0;
}
// For windows machine 
// cmake -G "Visual Studio 16 2019" .. ;  cmake --build . --config Release
